clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -475,6 +477,7 @@ private SchemaEntry maybeFetchValueSchemaEntryById(int valueSchemaId, boolean fo
         return NOT_EXIST_VALUE_SCHEMA_ENTRY;
       } else {
         valueSchemaEntryMap.put(valueSchemaId, entry);
+        shouldRefreshLatestValueSchemaEntry.compareAndSet(false, true);
```

Comment:
I am struggling to think if this is not necessary for now. But right now I think adding this will not introduce harm as even if forceRefresh is true and if it is intermediate error and this will only add one time latest schema query.